src/Batcave/main.ts

```
@@ -0,0 +1,785 @@
+
+        const chapterScript =
+            $(".page__chapters-list script")
+                .filter((_, el) => {
+                    const content = $(el).html();
+                    return content ? content.includes("__DATA__") : false;
+                })
+                .first()
+                .html() || "";
+
+        const jsonMatch = chapterScript.match(
+            /window\.__DATA__\s*=\s*({[\s\S]*?});/,
+        );
+        const jsonData = jsonMatch ? jsonMatch[1] : null;
+
+        interface ChapterData {
+            id: number;
+            title?: string;
+            posi: number;
+            date: string;
+        }
+
+        interface ParsedData {
+            chapters?: ChapterData[];
+        }
+
+        try {
+            if (!jsonData) throw new Error("No JSON data found");
+
+            const parsedData: ParsedData = JSON.parse(jsonData) as ParsedData;
+
+            if (parsedData.chapters) {
+                parsedData.chapters.forEach((chapter: ChapterData) => {
+                    if (chapter.id && typeof chapter.id === "number") {
+                        const [day, month, year] = chapter.date
+                            .split(".")
+                            .map(Number);
+                        const isoDate = `${year}-${month.toString().padStart(2, "0")}-${day.toString().padStart(2, "0")}`;
+
+                        chapters.push({
+                            chapterId: chapter.id.toString(),
+                            title: chapter.title || `Chapter ${chapter.posi}`,
+                            sourceManga,
+                            chapNum: chapter.posi,
+                            publishDate: new Date(isoDate),
+                            volume: 0,
+                            langCode: "🇬🇧",
+                        });
+                    } else {
+                        console.error(`Invalid Chapter`);
+                    }
+                });
+            }
+        } catch (err) {
```

You would want to throw an actual error instead of catching it and returning an empty list.